pkg/cmd/pr/comment/comment.go

```diff
@@ -69,21 +86,93 @@ func (o *Options) Run() error {
 	}
-	return o.commentPullRequest(pr)
+	return o.commentPullRequestWithStrategy(context.Background(), pr)
 }
 
-func (o *Options) commentPullRequest(pr *scm.PullRequest) error {
-	o.Result = pr
+func (o *Options) commentPullRequestWithStrategy(ctx context.Context, pr *scm.PullRequest) error {
+	switch o.Strategy {
+	case CreateCommentStrategy:
+		return o.create(ctx, pr)
+
+	case CreateIfNotExistCommentStrategy:
+		return o.createIfNotExists(ctx, pr)
+
+	case DeleteAndCreateCommentStrategy:
+		return o.deleteAndCreate(ctx, pr)
+
+	default:
+		return o.create(ctx, pr)
+	}
+}
 
-	ctx := context.Background()
+func (o *Options) create(ctx context.Context, pr *scm.PullRequest) error {
+	prName := "#" + strconv.Itoa(o.Number)
 	comment := &scm.CommentInput{Body: o.Comment}
 	_, _, err := o.ScmClient.PullRequests.CreateComment(ctx, o.FullRepositoryName, o.Number, comment)
-	prName := "#" + strconv.Itoa(o.Number)
 	if err != nil {
 		return errors.Wrapf(err, "failed to comment on pull request %s on repository %s", prName, o.FullRepositoryName)
 	}
 	log.Logger().Infof("commented on pull request %s on repository %s", prName, o.FullRepositoryName)
+	return nil
+}
+
+func (o *Options) list(ctx context.Context, pr *scm.PullRequest) ([]*scm.Comment, error) {
+	comments, _, err := o.ScmClient.PullRequests.ListComments(ctx, o.FullRepositoryName, o.Number, scm.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to list comments on pull request #%d on repository %s", o.Number, o.FullRepositoryName)
+	}
+
+	return comments, nil
+}
+
+func (o *Options) delete(ctx context.Context, pr *scm.PullRequest, comments []*scm.Comment) error {
+	prName := "#" + strconv.Itoa(o.Number)
+
+	for _, comment := range comments {
+		_, err := o.ScmClient.PullRequests.DeleteComment(ctx, o.FullRepositoryName, o.Number, comment.ID)
+		if err != nil {
+			return errors.Wrapf(err, "failed to delete comment with ID %d on pull request %s on repository %s", comment.ID, prName, o.FullRepositoryName)
+		}
+		log.Logger().Infof("deleted comment with ID %d on pull request %s on repository %s", comment.ID, prName, o.FullRepositoryName)
+	}
 
 	return nil
+}
+
+func (o *Options) createIfNotExists(ctx context.Context, pr *scm.PullRequest) error {
+	existingComments, err := o.list(ctx, pr)
+	if err != nil {
+		return err
+	}
+
+	for _, comment := range existingComments {
+		if comment.Body == o.Comment {
+			log.Logger().Infof("Similar comment already exists on pull request #%d on repository %s", o.Number, o.FullRepositoryName)
+			return nil
+		}
+	}
+
+	return o.create(ctx, pr)
+}
+
+func (o *Options) deleteAndCreate(ctx context.Context, pr *scm.PullRequest) error {
+	existingComments, err := o.list(ctx, pr)
+	if err != nil {
+		return err
+	}
+
+	similarComments := make([]*scm.Comment, 0)
+	for _, comment := range existingComments {
+		if comment.Body == o.Comment {
+			similarComments = append(similarComments, comment)
+		}
```

Comment:
```suggestion
	for i := range existingComments {
	        comment := existingComments[i]
		if comment.Body == o.Comment {
			similarComments = append(similarComments, comment)
		}
```

until go 1.22 and the "for loop fix" - see https://tip.golang.org/doc/go1.22 - you'll have to take care of the fact that the `comment` variable is shared

Reply:
thank you for the review :) !
fixed